editor/plugins/texture_editor_plugin.cpp

```diff
@@ -163,19 +222,48 @@ TexturePreview::TexturePreview(Ref<Texture2D> p_texture, bool p_show_metadata) {
 	checkerboard->set_texture_repeat(CanvasItem::TEXTURE_REPEAT_ENABLED);
 	centering_container->add_child(checkerboard);
 
+	{
+		Ref<Shader> shader;
+		shader.instantiate();
+		shader->set_code(texture_2d_shader);
```

Comment:
It would be nice to not compile the shader every time. Though fixing it would require static shader like in ColorPicker, so it's something for the future.
Also now that we have instance uniforms, the material can be shared too.

Reply:
Also applies to `Texture3DEditorPlugin` and `TextureLayeredEditorPlugin`

Reply:
Yep.

Reply:
@KoBeWi I did it, as suggested, same way as `ColorPicker` - https://github.com/godotengine/godot/pull/104488